Framework/PythonInterface/mantid/api/src/Exports/BinaryOperations.cpp

```diff
@@ -177,31 +178,17 @@ ResultType performBinaryOpWithDouble(const LHSType inputWS, const double value,
 
   // In order to recreate a history record of the final binary operation
   // there must be a record of the creation of the single value workspace used
-  // on the RHS here. This is achieved by running CreateSingleValuedWorkspace
-  // algorithm and adding the output workspace to the ADS. Adding the output
-  // to the ADS is critical so that workspace.name() is updated, by the ADS, to
-  // return the same string. WorkspaceProperty<TYPE>::createHistory() then
-  // records the correct workspace name for input into the final binary
-  // operation rather than creating a temporary name.
-  auto alg = API::AlgorithmManager::Instance().createUnmanaged(
-      "CreateSingleValuedWorkspace");
-  alg->setChild(false);
-  // we manually store the workspace as it's easier to retrieve the correct
-  // type from alg->getProperty rather than calling the ADS again and casting
-  alg->setAlwaysStoreInADS(false);
-  alg->initialize();
-  alg->setProperty<double>("DataValue", value);
+  // on the RHS here. This is achieved by using the workspace factory to create
+  // a WorkspaceSingleValue and adding it to the ADS. Adding the output to the
+  // ADS is critical so that workspace.name() is updated, by the ADS, to return
+  // the same string. WorkspaceProperty<TYPE>::createHistory() then records the
+  // correct workspace name for input into the final binary operation rather
+  // than creating a temporary name. Beam spreader transmission
+  MatrixWorkspace_sptr singleValue = std::dynamic_pointer_cast<MatrixWorkspace>(
```

Comment:
I'm afraid this is quite going to work as the history won't have a way of recreating the single value workspace. Consider something like:
```
s = CreateSampleWorkspace()
s *= 2
```

The script generated from the history with these changes looks like
```
CreateSampleWorkspace(OutputWorkspace='s')
Multiply(LHSWorkspace='s', RHSWorkspace='__python_binary_op_single_value', OutputWorkspace='s')
```
and is missing the line to create `__python_binary_op_single_value`. Before these changes the script generates:
```
CreateSampleWorkspace(OutputWorkspace='s')
CreateSingleValuedWorkspace(OutputWorkspace='__python_binary_op_single_value', DataValue=2)
Multiply(LHSWorkspace='s', RHSWorkspace='__python_binary_op_single_value', OutputWorkspace='s')
```

We could consider using Scale for these operations but this will be a fair bit of work that I don't think is high priority at the moment?

Reply:
Thanks for pointing this out. I was worried this could cause an issue. The binary operations appear to be the blocker in reworking/removing CreateSingleValuedWorkspaces. For now I've just reverted the changes in this file so the history is fully replicated.

I agree this is not high enough priority at the moment. The crash caused by interacting with the SingleValuedWorkspace in Workbench has been fixed, so I guess for now the SingleValuedWorkspaces can stick around.